### PR TITLE
[FIX] on reconnect wait for server to be synced

### PR DESCRIPTION
**Pull request by darkdarkdragon**
on reconnect do not emit connected event till server_state is full

handle websocket errors in browsers

### CODE REVIEW

File: src/common/connection.js [hunk 1]
```diff
@@ -162,15 +169,27 @@ class Connection extends EventEmitter {
       }
 
       this._updateLedgerVersions(data);
+      return this.request({command: 'server_info'});
+    }).then(serverInfo => {
+      const serverState = _.get(serverInfo, 'info.server_state');
+      if (serverState !== 'full') {
+        // rippled instance still syncing, so it can't be used
+        return this._disconnect(false).then(() => {
+          throw new NotConnectedError('Rippled is syncing');
```
Comment (clark800) [inline src/common/connection.js]: How about `RippledNotSyncedError` to distinguish it?

**Comment by clark800** [comment]
The connection class is getting more and more complicated, is there any refactoring that can be done to make it more manageable?

File: src/common/connection.js (hunk 1 above)
Comment (darkdarkdragon) [inline src/common/connection.js]: Added

**Comment by darkdarkdragon** [comment]
yes, I was thinking about that. One thing I can think out - it is to move all connect/reconnect/error handling logic into base class or to separate ws wrapper.

**Comment by clark800** [comment]
Yeah, I was thinking along the same lines: a `ws` wrapper that separates out the error handling and all the event handlers. Would you be up for working on that?

**Comment by darkdarkdragon** [comment]
@clark800 fixed

**Comment by darkdarkdragon** [comment]
@clark800 done

**Comment by clark800** [comment]
LGTM

**Pull request merged**